sample_metadata/parser/generic_parser.py

```diff
@@ -26,30 +42,73 @@
     AnalysisStatus,
 )
 
+
+# https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-new-additions-to-the-typing-module
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
 logger = logging.getLogger(__file__)
 logger.setLevel(logging.INFO)
 
 FASTQ_EXTENSIONS = ('.fq', '.fastq', '.fq.gz', '.fastq.gz')
 BAM_EXTENSIONS = ('.bam',)
 CRAM_EXTENSIONS = ('.cram',)
-VCFGZ_EXTENSIONS = ('.vcf.gz',)
+GVCF_EXTENSIONS = ('.g.vcf.gz',)
+VCF_EXTENSIONS = ('.vcf', '.vcf.gz')
+
+ALL_EXTENSIONS = FASTQ_EXTENSIONS + CRAM_EXTENSIONS + BAM_EXTENSIONS + GVCF_EXTENSIONS + VCF_EXTENSIONS
 
 rmatch = re.compile(r'[_\.-][Rr]\d')
 GroupedRow = Union[List[Dict[str, Any]], Dict[str, Any]]
 
+T = TypeVar('T')
+
+SUPPORTED_FILE_TYPE = Literal['reads', 'variants']
+SUPPORTED_READ_TYPES = Literal['fastq', 'bam', 'cram']
+SUPPORTED_VARIANT_TYPES = Literal['gvcf', 'vcf']
+
+
+def chunk(iterable: Sequence[T], chunk_size=500) -> Iterator[Sequence[T]]:
+    """
+    Chunk a sequence by yielding lists of `chunk_size`
+    """
+    for i in range(0, len(iterable), chunk_size):
+        yield iterable[i: i + chunk_size]
+
+
+def run_as_sync(f):
+    """
+    Run an async function, synchronously.
+    Useful for @click functions that must be async, eg:
+
+    @click.command()
+    @click.option(...)
+    @run_as_sync
+    async def my_async_function(**kwargs):
+        return await awaitable_function(**kwargs)
+    """
+
+    def wrapper(*args, **kwargs):
+        loop = asyncio.get_event_loop()
```

Comment:
It's a good practice to add a `wraps` decorator to the inner wrapper, to make the decorated function inherit attributes of the original function: https://docs.python.org/3/library/functools.html#functools.wraps

```suggestion
    def wrapper(*args, **kwargs):
        @functools.wraps(f)
        loop = asyncio.get_event_loop()
```